### PR TITLE
Cut Prometheus 0.16.0rc1.

### PR DESCRIPTION
**Pull request by juliusv**

### CODE REVIEW

**Comment by juliusv** [comment]
@fabxc @beorn7 @brian-brazil

**Comment by brian-brazil** [comment]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,139 @@
...
+  they are only appended when communicating with external systems
+  (Alertmanager, remote storages, federation).
+* The names and units of metrics related to remote storage sample appends have
+  been changed.
+* The experimental support for writing to InfluxDB has been updated to work
+  with InfluxDB 0.9.x. 0.8.x versions of InfluxDB are not supported anymore.
+
+All changes:
+
+* [CHANGE] Vendoring is now done via govendor instead of godep.
```
Comment (beorn7) [inline CHANGELOG.md]: Perhaps mention that we require Go 1.5.x now and the GO15VENDOREXPERIMENT=1 env variable.

Comment (juliusv) [inline CHANGELOG.md]: Done.

Comment (fabxc) [inline CHANGELOG.md]: Using `make` the latter is not a user requirement really.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,139 @@
...
+  native Go build mechanisms and new Go 1.5 experimental vendoring support.
+* [CLEANUP] Remove unused .pkgignore file.
+* [CLEANUP] Remove logged notice about 0.14.x configuration file format change.
+* [CLEANUP] Move scrape-time metric label modification into SampleAppenders.
+* [CLEANUP] Switch from `github.com/client_golang/model` to
+  `github.com/common/model` and related type cleanups.
+* [CLEANUP] Switch from `github.com/client_golang/extraction` to
+  `github.com/common/expfmt` and related type cleanups.
+* [CLEANUP] Exit Prometheus when the web server encounters a startup error.
+* [CLEANUP] Simplify Travis CI configuration.
```
Comment (fabxc) [inline CHANGELOG.md]: Stuff like that has nothing to do with the behavior of the Prometheus application. Personally, I wouldn't list such things here (similarly the .pkgignore thing) – the list is nosiy enough already.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,139 @@
...
+  backlogging.
+* [ENHANCEMENT] Make `hash_mod` relabeling action use MD5 instead of FNV to
+  enable better hash distribution.
+* [ENHANCEMENT] Better tracking of targets between same service discovery
+  mechanisms in one scrape configuration.
+* [ENHANCEMENT] Handle parser and query evaluation runtime panics more
+  gracefully.
+* [ENHANCEMENT] Add IDs to H2 tags on status page to allow anchored linking.
+* [BUGFIX] Fix series file deletion behavior when purging archived series.
+* [BUGFIX] Fix scraper HTTP connection leak when encountering HTTP errors.
```
Comment (fabxc) [inline CHANGELOG.md]: That was never in 0.15.0 – we shouldn't list fixing of errors we introduced in the development phase.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,139 @@
...
+  proxy servers.
+* [FEATURE] Add console templates for Prometheus itself.
+* [FEATURE] Allow relabeling the protocol scheme of targets.
+* [FEATURE] Add `predict_linear()` query language function.
+* [FEATURE] Support for authentication using bearer tokens, client certs, and
+  CA certs.
+* [FEATURE] Implement unary expressions for vector types (`-foo`, `+foo`).
+* [FEATURE] Add console templates for the SNMP exporter.
+* [FEATURE] Make it possible to relabel target scrape query parameters.
+* [FEATURE] Add support for `A` records in DNS service discovery.
```
Comment (fabxc) [inline CHANGELOG.md]: `A` and `AAAA`

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,139 @@
...
+* [ENHANCEMENT] Update the architecture diagram in the `README.md`.
+* [ENHANCEMENT] Time out sample appends in retrieval layer if the storage is
+  backlogging.
+* [ENHANCEMENT] Make `hash_mod` relabeling action use MD5 instead of FNV to
+  enable better hash distribution.
+* [ENHANCEMENT] Better tracking of targets between same service discovery
+  mechanisms in one scrape configuration.
+* [ENHANCEMENT] Handle parser and query evaluation runtime panics more
+  gracefully.
+* [ENHANCEMENT] Add IDs to H2 tags on status page to allow anchored linking.
```
Comment (fabxc) [inline CHANGELOG.md]: Noise.

File: CHANGELOG.md (hunk 4 above)
Comment (juliusv) [inline CHANGELOG.md]: Fixing.

File: CHANGELOG.md (hunk 5 above)
Comment (juliusv) [inline CHANGELOG.md]: I'll clarify that this is actually a user-requested feature (to be able to link to them).

File: CHANGELOG.md (hunk 3 above)
Comment (juliusv) [inline CHANGELOG.md]: Removing.

File: CHANGELOG.md (hunk 2 above)
Comment (juliusv) [inline CHANGELOG.md]: Don't feel strongly about this, but can remove.

**Comment by fabxc** [comment]
:+1:

File: CHANGELOG.md (hunk 5 above)
Comment (juliusv) [inline CHANGELOG.md]: Oh, actually it is already clarified in the text :)

**Comment by juliusv** [comment]
Addressed all comments again (also removed the vendoring env var explanation again, as we also didn't explain how to build with vendored deps besides `make build` before).

**Comment by beorn7** [comment]
Fair enough, without GO15VENDOREXPERIMENT=1 you can build against your checked out dependencies. Still I think its worth mentioning that we have `vendor` directories now, so the behavior of `go build` changes. That's something that should be mentioned as it is now part of the go tool.

**Comment by beorn7** [comment]
Thanks for listening to me... :-(

**Comment by juliusv** [comment]
Ok, let's do that with the next RC or full release.

**Pull request merged**